File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/exception/LenderOperationException.java
package com.nivasafinance.features.lender.lender.exception;

import org.springframework.context.MessageSource;

import java.io.Serial;

public class LenderOperationException extends RuntimeException {

    @Serial
    private static final long serialVersionUID = 1L;

    public LenderOperationException(String messageKey, MessageSource messageSource) {
        super(com.nivasafinance.common.exception.ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource));
    }
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/exception/LenderOperationException.kt
package com.nivasafinance.features.lender.lender.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource

class LenderOperationException(
    messageKey: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource)
)
